public/pages/workflow_detail/workspace/resizable_workspace.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useRef, useState } from 'react';
+import { ReactFlowProvider } from 'reactflow';
+import { EuiResizableContainer } from '@elastic/eui';
+import { Workflow } from '../../../../common';
+import { Workspace } from './workspace';
+import { ComponentDetails } from './component_details';
+
+interface ResizableWorkspaceProps {
+  workflow?: Workflow;
+}
+
+/**
+ * The overall workspace component that maintains state related to the 2 resizable
+ * panels - the ReactFlow workspace panel and the selected component details panel.
+ */
+export function ResizableWorkspace(props: ResizableWorkspaceProps) {
+  const [isOpen, setIsOpen] = useState<boolean>(true);
+  const collapseFn = useRef(
+    (id: string, options: { direction: 'left' | 'right' }) => {}
+  );
+
+  const onToggleChange = () => {
+    collapseFn.current('inputsPanel', { direction: 'left' });
+    setIsOpen(!isOpen);
+  };
+
+  return (
+    <EuiResizableContainer
+      direction="horizontal"
+      style={{ marginLeft: '-14px' }}
+    >
+      {(EuiResizablePanel, EuiResizableButton, { togglePanel }) => {
+        if (togglePanel) {
+          collapseFn.current = (panelId: string, { direction }) =>
+            togglePanel(panelId, { direction });
+        }
+
+        return (
+          <ReactFlowProvider>
```

Comment:
Additional clarification question:

From my understanding a React `context` is used so the parent component make some information available to any component in the tree below it. We have our `ReactFlowContextProvider`, does this have the info on all reactflowInstances? 

For example when we have the component details and workspace wrapped in a reactflowprovider it means each workflow will have a provider to get its own rfInstance? otherwise this line `const { reactFlowInstance } = useContext(rfContext);` in `[component_details]` (https://github.com/opensearch-project/opensearch-ai-flow-dashboards/pull/57/files#diff-8c84fba8c626d0c26f252d1c729485bf1401aa7ac15ba8015fe942d02ff32c2fR37) wont know which reactflow instance to get right?

Additioonally you have the EuiResizableContainer as a parent of ReactFlowProvider. why not the other way around?

Reply:
Both `ReactFlowContextProvider` and `ReactFlowProvider` are only intended for a single reactflow instance. We reactflow instance is only set once in `Workspace` via `onInit()` fn provided by reactflow.

To clarify the 2 different uses:
- `ReactFlowContextProvider` is for accessing the instance anywhere within the plugin - see details in #42 
- `ReactFlowProvider` is for accessing the custom hooks provided by reactflow to listen for certain actions taken within the workspace - see this PR description for details

Reply:
The reason for putting the ReactFlowProvider where I did was just to try to scope it as small as possible to prevent weird side effects.

Reply:
got it so the only place where we have knowledge of all the workflows together is through redux state right?

Reply:
Yes - workflows are just the standalone backend resources, and we only persist one reactflow instance when a particular workflow is loaded into the details page.